k8s/scripts/inspect.sh

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env bash
+
+INSPECT_DUMP=$(pwd)/inspection-report
+
+function log_success {
+  printf -- '\033[32m SUCCESS: \033[0m %s\n' "$1"
+}
+
+function log_info {
+  printf -- '\033[34m INFO: \033[0m %s\n' "$1"
+}
+
+function log_warning () {
+  printf -- '\033[33m WARNING: \033[0m %s\n' "$1"
+}
+
+function log_warning_red {
+  printf -- '\033[31m WARNING: \033[0m %s\n' "$1"
+}
+
+function is_control_plane_node {
+  k8s local-node-status | grep -q "control-plane"
+}
+
+function collect_args {
+  log_info "Copy service args to the final report tarball"
+  cp -r --no-preserve=mode,ownership /var/snap/k8s/common/args "$INSPECT_DUMP"
+}
+
+function collect_cluster_info {
+  log_info "Copy k8s cluster-info dump to the final report tarball"
+  k8s kubectl cluster-info dump &> "$INSPECT_DUMP/cluster-info.log"
+}
+
+function collect_sbom {
+  log_info "Copy SBOM to the final report tarball"
+  cp --no-preserve=mode,ownership /snap/k8s/current/bom.json "$INSPECT_DUMP"/sbom.json
+}
+
+function collect_diagnostics {
+  log_info "Copy uname to the final report tarball"
+  uname -a &>"$INSPECT_DUMP/uname.log"
+
+  log_info "Copy snap diagnostics to the final report tarball"
+  snap version &>"$INSPECT_DUMP/snap-version.log"
+  snap list k8s &>"$INSPECT_DUMP/snap-list-k8s.log"
+  snap services k8s &>"$INSPECT_DUMP/snap-services-k8s.log"
+  snap logs k8s -n 10000 &>"$INSPECT_DUMP/snap-logs-k8s.log"
+
+  log_info "Copy k8s diagnostics to the final report tarball"
+  k8s kubectl version &>"$INSPECT_DUMP/k8s-version.log"
+  k8s status &>"$INSPECT_DUMP/k8s-status.log"
+  k8s get &>"$INSPECT_DUMP/k8s-get.log"
+  k8s kubectl get cm k8sd-config -n kube-system -o yaml &>"$INSPECT_DUMP/k8sd-configmap.log"
+  k8s kubectl get cm -n kube-system &>"$INSPECT_DUMP/k8s-configmaps.log"
+
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/cluster.yaml "$INSPECT_DUMP/k8s-dqlite-cluster.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8s-dqlite/info.yaml "$INSPECT_DUMP/k8s-dqlite-info.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/cluster.yaml "$INSPECT_DUMP/k8sd-cluster.yaml"
+  cp --no-preserve=mode,ownership /var/snap/k8s/common/var/lib/k8sd/state/database/info.yaml "$INSPECT_DUMP/k8sd-info.yaml"
+}
+
+function check_service {
+  local service
+  service=$1
+
+  mkdir -p "$INSPECT_DUMP/$service"
+
+  local status_file
+  status_file="$INSPECT_DUMP/$service/systemctl.log"
+
+  systemctl status "snap.$service" &> "$status_file"
+
+  if grep -q "active (running)" "$status_file"; then
+    log_info "Service $service is running"
+  else
+    log_info "Service $service is not running"
+  fi
+
+  printf -- "%s $(systemctl show "snap.$service" -p NRestarts)\n" "$service" >> "$INSPECT_DUMP/nrestarts.log"
+
+  journalctl -n 100000 -u "snap.$service" &>"$INSPECT_DUMP/$service/journal.log"
+}
```

Comment:
we need a warning if the service is not "healthy" but we expect it to be. e.g. worker nodes need the apiserver-proxy service to be up. control plane nodes need the control plane services to be up. all nodes need kubelet, containerd, etc